diff -ur test1/bcm430x.c test1.ahi/bcm430x.c
--- test1/bcm430x.c	2005-08-23 18:44:18.000000000 -0700
+++ test1.ahi/bcm430x.c	2005-08-23 19:09:10.000000000 -0700
@@ -119,56 +119,56 @@
 	bcm430x_write32(bcm, val, BCM430x_SHM_DATA);
 }
 
-static int bcm430x_pci_read_config_8(struct pci_dev *pdev, u16 offset, u8 val)
+static int bcm430x_pci_read_config_8(struct pci_dev *pdev, u16 offset, u8 *val)
 {
-	u8 tmp; int err;
+	int err;
 	
-	err = pci_read_config_byte(pdev, offset, &tmp);
-	printk(KERN_INFO "bcm430x:  read 0x%04x  0x%02x\n", offset, tmp);
+	err = pci_read_config_byte(pdev, offset, val);
+	printk(KERN_INFO "bcm430x:  read 0x%04x  0x%02x\n", offset, *val);
 	return err;
 }
 
-static int bcm430x_pci_read_config_16(struct pci_dev *pdev, u16 offset, u16 val)
+static int bcm430x_pci_read_config_16(struct pci_dev *pdev, u16 offset, u16 *val)
 {
-	u16 tmp; int err;
+	int err;
 	
-	err = pci_read_config_word(pdev, offset, &tmp);
-	printk(KERN_INFO "bcm430x:  read 0x%04x  0x%04x\n", offset, tmp);
+	err = pci_read_config_word(pdev, offset, val);
+	printk(KERN_INFO "bcm430x:  read 0x%04x  0x%04x\n", offset, *val);
 	return err;
 }
 
-static int bcm430x_pci_read_config_32(struct pci_dev *pdev, u16 offset, u32 val)
+static int bcm430x_pci_read_config_32(struct pci_dev *pdev, u16 offset, u32 *val)
 {
-	u32 tmp; int err;
+	int err;
 	
-	err = pci_read_config_dword(pdev, offset, &tmp);
-	printk(KERN_INFO "bcm430x:  read 0x%04x  0x%08x\n", offset, tmp);
+	err = pci_read_config_dword(pdev, offset, val);
+	printk(KERN_INFO "bcm430x:  read 0x%04x  0x%08x\n", offset, *val);
 	return err;
 }
 
-static int bcm430x_pci_write_config_8(struct pci_dev *pdev, u16 offset, u8 val)
+static int bcm430x_pci_write_config_8(struct pci_dev *pdev, int offset, u8 val)
 {
-	printk(KERN_INFO "bcm430x:  write 0x%04  0x%02x\n", offset, val);
-	return pci_write_config_byte(pdev, offset, &val);
+	printk(KERN_INFO "bcm430x:  write 0x%04x  0x%02x\n", offset, val);
+	return pci_write_config_byte(pdev, offset, val);
 }
 
-static int bcm430x_pci_write_config_16(struct pci_dev *pdev, u16 offset, u16 val)
+static int bcm430x_pci_write_config_16(struct pci_dev *pdev, int offset, u16 val)
 {
-	printk(KERN_INFO "bcm430x:  write 0x%04  0x%04x\n", offset, val);
-	return pci_write_config_word(pdev, offset, &val);
+	printk(KERN_INFO "bcm430x:  write 0x%04x  0x%04x\n", offset, val);
+	return pci_write_config_word(pdev, offset, val);
 }
 
-static int bcm430x_pci_write_config_32(struct pci_dev *pdev, u16 offset, u32 val)
+static int bcm430x_pci_write_config_32(struct pci_dev *pdev, int offset, u32 val)
 {
-	printk(KERN_INFO "bcm430x:  write 0x%04  0x%08x\n", offset, val);
-	return pci_write_config_dword(pdev, offset, &val);
+	printk(KERN_INFO "bcm430x:  write 0x%04x  0x%08x\n", offset, val);
+	return pci_write_config_dword(pdev, offset, val);
 }
 
 /* Read SPROM and fill the useful values in the net_device struct */
 static void bcm430x_read_sprom(struct net_device *dev)
 {
 	struct bcm430x_private *bcm = netdev_priv(dev);
-	u16 mac1, mac2, mac3;
+	int mac1, mac2, mac3;
 	
 	/* Deal with short offsets */
 	#define READ_SPROM(addr) le16_to_cpu(bcm430x_read16(bcm, 0x1000 + 2*(addr)))
@@ -179,9 +179,9 @@
 
 	#undef READ_SPROM
 
-	memcpy (dev->dev_addr+0, mac3, 2);
-	memcpy (dev->dev_addr+2, mac2, 2);
-	memcpy (dev->dev_addr+4, mac1, 2);
+	memcpy (dev->dev_addr+0, &mac3, 2);
+	memcpy (dev->dev_addr+2, &mac2, 2);
+	memcpy (dev->dev_addr+4, &mac1, 2);
 
 	printk(KERN_INFO
 	       "bcm430x: set MAC address %04x%04x%04x\n", mac1, mac2, mac3);
@@ -274,7 +274,7 @@
 	bcm = netdev_priv(dev);
 	bcm->pci_dev = pdev;
 
-	err = pci_enable_device(dev);
+	err = pci_enable_device(pdev);
 	if (err) {
 		printk(KERN_ERR
 		       "bcm430x: unable to wake up pci device (%i)\n", err);
@@ -366,8 +366,9 @@
 	bcm = netdev_priv(dev);
 	ioaddr = bcm->mmio_addr;
 
-	bcm430x_read_sprom(&dev);
+	bcm430x_read_sprom(dev);
 
+	return 0;
 }
 
 static void __devexit bcm430x_remove_one (struct pci_dev *pdev)
@@ -387,10 +388,12 @@
 
 static int bcm430x_suspend (struct pci_dev *pdev, pm_message_t state)
 {
+	return 0;
 }
 
 static int bcm430x_resume (struct pci_dev *pdev)
 {
+	return 0;
 }
 
 #endif /* CONFIG_PM */
@@ -406,7 +409,7 @@
 #endif /* CONFIG_PM */
 };
 
-static void __init bcm430x_init(void)
+static int __init bcm430x_init(void)
 {
 #ifdef MODULE
 	printk(KERN_INFO BCM430x_DRIVER_NAME "\n");
